libs/sdk-ui-dashboard/src/model/commandHandlers/drillTargets/addDrillTargetsHandler.ts

```diff
@@ -3,41 +3,32 @@ import { SagaIterator } from "redux-saga";
 import { put, select } from "redux-saga/effects";
 import { AddDrillTargets } from "../../commands/drillTargets";
 import { DrillTargetsAdded, drillTargetsAdded } from "../../events/drillTargets";
-import { invalidArgumentsProvided } from "../../events/general";
 import { drillTargetsActions } from "../../state/drillTargets";
 import { DashboardContext } from "../../types/commonTypes";
 import { selectWidgetsMap } from "../../state/layout/layoutSelectors";
-import { objRefToString } from "@gooddata/sdk-model";
+import { validateExistingInsightWidget } from "../widgets/validation/widgetValidations";
 
 export function* addDrillTargetsHandler(
     ctx: DashboardContext,
     cmd: AddDrillTargets,
 ): SagaIterator<DrillTargetsAdded> {
     const {
-        payload: { widgetRef, availableDrillTargets },
+        payload: { availableDrillTargets },
         correlationId,
     } = cmd;
-    const existingWidgets: ReturnType<typeof selectWidgetsMap> = yield select(selectWidgetsMap);
-    const widget = existingWidgets.get(widgetRef);
 
-    if (!widget) {
-        throw invalidArgumentsProvided(
-            ctx,
-            `Attempting to register drill targets for non-existing widget with ref ${objRefToString(
-                widgetRef,
-            )}. The drill targets will be ignored.`,
-            correlationId,
-        );
-    }
+    const widgets: ReturnType<typeof selectWidgetsMap> = yield select(selectWidgetsMap);
+    const insightWidget = validateExistingInsightWidget(widgets, cmd, ctx);
+    const { ref, uri, identifier } = insightWidget;
 
     yield put(
         drillTargetsActions.addDrillTargets({
-            identifier: widget.identifier,
-            uri: widget.uri,
-            ref: widgetRef,
+            identifier: identifier,
+            uri: uri,
+            ref: ref,
```

Comment:
Nipick: you can now use shorthand properties.

Reply:
Done